chore: adds reactJS

caminoezequiel opened the pull request:
Link to Test: https://play.decentraland.zone/branch/chore/adds_react/index.html?ENV=zone

Commit:
replacing html element for redux events

Commit:
Adds Web3Connector and its provider

Commit:
updates sign-in flow and web3 connection

Commit:
Updates loading sagas to user reactjs

Commit:
adds reacts components

Commit:
config react

Commit:
Updates makefile to build react app on watch

Commit:
adds errors components

Commit:
Adds details for network mismatch

Commit:
update webpack to process css

Commit:
fixing lint errors

Commit:
add react build to makefile

Commit:
removing console log

Commit:
add website as create react app

Commit:
Removing react from kernel

Commit:
remove old version of website in kernel

Commit:
adds storybook test

Commit:
fixes and updates snapshots test

Commit:
moving stories together with components

Commit:
adds symbols link to loader to allow to test on localhost

Commit:
updates styles

Commit:
revert kernel, now we'll use react app

Commit:
removing getElementById

Commit:
updates website snapshots test

Commit:
adds warning messages for preview mode

Commit:
ordering react app

Commit:
adds decentraland-kernel

Commit:
Merge remote-tracking branch 'origin/master' into chore/adds_react

Commit:
updates unity entrypoint to export init functions

Commit:
adds main file to decentraland-kernel

Commit:
updates init from decentraland-kernel

Commit:
adds website entrypoint, revert unity entry point

Commit:
adds post install to copy decentraland-kernel files

Commit:
implements website entry point in react

Commit:
add html for preview

Commit:
adds react-app to circleci

Commit:
Merge remote-tracking branch 'origin/master' into chore/adds_react

Commit:
fixing lint alert

Commit:
adds website to circleci workspace

Commit:
updates circleci

Commit:
Update circleci

Commit:
removing serviceWorker

Commit:
updates circleci

Commit:
updates circleci

Commit:
adds npm link to website step

Commit:
updates cicleci -> install decentraland-kernel from file

Commit:
Updates circleci

Commit:
Updates circleci

Commit:
Update circleci

Commit:
Update config.yml

Commit:
Update config.yml

Commit:
updates circleci

menduz committed:
use built kernel using CP

Commit:
adds explorer version, pre-install and post-install scripts

Commit:
adds guest mode

Commit:
updates website post-install scritps and circleci

Commit:
fixing lint errors

Commit:
updates post-install scripts

Commit:
changes loadingContainer to show images only on initialLoading

Commit:
Adds fortmatic to kernel

Commit:
updates pre/post scripts

pravusjif reviewed (approved):
Just tested the build, and had no issues.

Commit:
Adds dafault catalyst

Commit:
Merge remote-tracking branch 'origin/master' into feat/signIn_signUp

Commit:
Adds wallet images

Commit:
adds website to make watch

Commit:
new flow singIn

Commit:
draft sign-up on kernel

Commit:
adds signup components

Commit:
Merge remote-tracking branch 'origin/master' into feat/signIn_signUp

menduz commented on kernel/Makefile:
```diff
@@ -260,13 +260,14 @@ lint-fix: ## Fix bad formatting on all .ts and .tsx files
 			"$(COMPILER) targets/engine/loader.json --watch" \
 			"$(COMPILER) targets/scenes/basic-scenes.json --watch" \
 			"$(COMPILER) targets/entryPoints/unity.json --watch" \
+			"$(COMPILER) targets/entryPoints/website.json --watch" \
```
is this still valid?

caminoezequiel replied:
@menduz Yes, it is. Watching website.json allows us see the updates of kernel in reactjs when we are in local

Commit:
Merge remote-tracking branch 'origin/master' into chore/adds_react

Commit:
draft show avatar editor

Commit:
chore: renderer modifications for new sign up

Commit:
Merge branch 'chore/adds_react' of https://github.com/decentraland/explorer into chore/renderer-new-sigin-flow

Commit:
adds signUp flow

Commit:
updates components and test

Commit:
Merge remote-tracking branch 'origin/master' into chore/adds_react

Commit:
fix userId for signUp

Commit:
disable terms of services flag

Commit:
Merge remote-tracking branch 'origin/master' into chore/adds_react

Commit:
activate rendering after signUp and set tutorial step 0

Commit:
updates reactjs tests

Commit:
updates styles

Commit:
updates styles

Commit:
update background

pablitar commented on kernel/packages/shared/Html.ts:
```diff
@@ -0,0 +1,189 @@
+import { LoadingState } from './loading/reducer'
```
Love having all this nasty stuff in the same place!

Commit:
updates button color, adds form validation

Commit:
fix onChange type

Commit:
updates logo and organizes components

Commit:
updates react test

Commit:
migrates some images to reacts, changes eth login

Commit:
updates react test

Commit:
updates styles

Commit:
updates react test, adds disabled agree until user scroll to the end

Commit:
updates loading on home pages

menduz reviewed (approved):
Looks good, resolve the conflicts and ship it :shipit:

Commit:
Adds avatar face for passport step

Commit:
updates session sagas

pablitar commented on website/src/components/auth/LoginHeader.tsx:
```diff
@@ -0,0 +1,11 @@
+  <div className="eth-login-description">
+    <Logo />
+    <p>
+      Increase yourself into the first virtual world fully owned by its users.
```
```suggestion
      Inmerse yourself into the first virtual world fully owned by its users.
```

pablitar commented on kernel/packages/shared/dao/sagas.ts:
```diff
@@ -41,12 +41,51 @@ const CATALYST_CANDIDATES_KEY = CACHE_KEY + '-' + SET_CATALYST_CANDIDATES
   yield takeEvery([SET_CATALYST_CANDIDATES, SET_ADDED_CATALYST_CANDIDATES], cacheCatalystCandidates)
 }
 
+function* loadDefaultCatalystRealms() {
```
I suppose we need this to be able to run the avatar editor, right? isn't there another workaround?

caminoezequiel replied:
It's necessary to kwon if a profile already exists or not. Today the realm is initialized after calling `loginCompleted` but now we need to kwon if the address has a profile before `loginCompleted` to kwon if user need to continue with the sign up or sing in flow.

pablitar commented on kernel/packages/shared/ethereum/Web3Connector.ts:
```diff
@@ -0,0 +1,52 @@
+import { ProviderType } from './ProviderType'
+import { ConnectorInterface } from './connector/ConnectorInterface'
+
+export class Web3Connector {
```
This is really nice

pablitar commented on kernel/packages/shared/ethereum/provider.ts:
```diff
@@ -1,19 +1,13 @@
-import { WebSocketProvider, RequestManager } from 'eth-connect'
```
Love all the deleted lines here

pablitar commented on kernel/packages/shared/index.ts:
```diff
@@ -9,29 +9,33 @@ import { loadingStarted, notStarted, MOBILE_NOT_SUPPORTED } from './loading/type
 declare const globalThis: StoreContainer
 
 export function initShared() {
+  if (globalThis.globalStore) {
```
does this mean that now initShared can be called multiple times?

caminoezequiel replied:
No, it is called only one time, the difference is that in react app we need to have the redux store before `initShared` is called then here we check it.

pablitar commented on kernel/packages/shared/profiles/sagas.ts:
```diff
@@ -387,15 +403,26 @@ function* populateFaceIfNecessary(profile: any, resolution: string) {
+  return request.json()
+}
+
+export function* createSignUpProfile(profile: Profile, identity: ExplorerIdentity) {
```
We already talked about this, but ensure to use saveProfileRequest so we minimize alternative flows!

pablitar commented on kernel/packages/shared/session/sagas.ts:
```diff
@@ -1,98 +1,267 @@
-      // @ts-ignore
-      originalOnChange && originalOnChange(e)
-    }
+function* initialize() {
```
Nice

pablitar commented on kernel/packages/shared/session/sagas.ts:
```diff
@@ -143,12 +312,12 @@ function* login() {
   yield put(userAuthentified(userId, identity, net))
 
-  yield loginCompleted
+  loginCompleted.resolve()
```
Just a sanity check: `yield loginCompleted` awaited for the background login to finish... I think.

This completely changes that behaviour to marking the login completed when this line is hit.

Is this the expected behaviour?

We have a couple of `loginCompleted.resolve()` above

caminoezequiel replied:
Yes, i think so. Let me check why it changed and i'll talk you later

pablitar commented on kernel/packages/unity-interface/BrowserInterface.ts:
```diff
@@ -178,11 +183,25 @@ export class BrowserInterface {
+      globalThis.globalStore.dispatch(signUpSetProfile(updated))
+      globalThis.globalStore.dispatch(changeSignUpStage('passport'))
+      unityInterface.DeactivateRendering()
+      document.getElementById('gameContainer')!.setAttribute('style', 'display: none')
```
Can't you put this in the `HTML` module that you've created?

caminoezequiel replied:
yes! of course, I've seen that line duplicated in so many places

pablitar reviewed (approved):
This is huge. Great work!

Please review the comments before merging, but no need to block the PR because of them.

We may have to fix some things, but once this thing is in master, we can work much faster an in parallel to make it stable :)

Commit:
Merge remote-tracking branch 'origin/master' into chore/adds_react

Commit:
Adds loadings, adds go to avatar editor, save last address to don't show wallet selector

Commit:
lint fixing

Commit:
fixing identity

Commit:
remove teleport animation for initial load

Commit:
Merge remote-tracking branch 'origin/master' into chore/adds_react

Commit:
wording

Co-authored-by: pablitar <pablo@decentraland.org>

pablitar committed:
feat: Unclaimed names support (#1544)

Commit:
Merge remote-tracking branch 'origin/master' into chore/adds_react

Commit:
wording

Commit:
Merge remote-tracking branch 'origin/master' into chore/adds_react

Commit:
changes pre start script to work in windows

Commit:
Change baseUrl to run unity on localhost

Commit:
wording

Commit:
now guest need to complete avatar editor

Commit:
Merge remote-tracking branch 'origin/master' into chore/adds_react

Commit:
Adds cancel signUp in kernel

Commit:
Adds cancel signUp

Commit:
adds Fortmatic keys

Commit:
adds guest sign up flow

Commit:
adds email validation to passport form

Commit:
send profile email to analytics

Commit:
Merge remote-tracking branch 'origin/master' into chore/adds_react

2fd committed:
fix: add new styles

2fd committed:
fix: optimaze images

2fd committed:
fix: small device styles

Commit:
Merge remote-tracking branch 'origin/master' into chore/adds_react

2fd committed:
fix: lint

2fd committed:
fix: lint

2fd committed:
fix: lint

2fd committed:
fix: lint

2fd committed:
fix: unity styles

2fd committed:
fix: popup styles

2fd committed:
fix: styles

menduz commented on .circleci/config.yml:
```diff
@@ -270,6 +277,45 @@ jobs:
+          command: sh scripts/post-install.sh
+      - run:
+          name: "creating .env (ignore master)"
+          command: echo "PUBLIC_URL=\"https://play.decentraland.zone/branch-website/${CIRCLE_BRANCH}\"" >> .env
```
This won't work in .today or .org

caminoezequiel replied:
@menduz Do you know how we can set the top level domain dynamically? can we know it from the branch name?

menduz replied:
we have no way to do it at any build stage of building currently, as they are branch agnostic... is there any chance to not require the public_url to generate the artifacts?

caminoezequiel replied:
it's require only for branches because we are deploying on a sub directory so we can use absolute paths. It's the reason why I ignore `master` and `tags` on lines 301 to 303

menduz commented on kernel/packages/build-ecs/index.ts:
```diff
@@ -162,11 +162,11 @@ async function minify(files: string | string[] | { [file: string]: string }) {
+    //   ecma: 5,
+    //   comments: /^!/,
+    //   beautify: false
+    // },
```
Please restore this, it is **_REALLY_** necessary

caminoezequiel replied:
yes!, of course. I don't know why these lines are comment, maybe @2fd knows

menduz commented on kernel/packages/shared/ethereum/connector/ConnectorFactory.ts:
```diff
@@ -0,0 +1,30 @@
+import { ETHEREUM_NETWORK, ethereumConfigurations } from '../../../config'
+import { FortmaticConnector } from './FortmaticConnector'
+
+export class ConnectorFactory {
```
this class can be a simple function

menduz commented on kernel/packages/shared/session/actions.ts:
```diff
@@ -15,8 +27,48 @@ export type UserAuthentified = ReturnType<typeof userAuthentified>
 export const logout = () => action(LOGOUT)
 export type Logout = ReturnType<typeof logout>
+
+export const UPDATE_TOS = 'UPDATE_TOS'
```
missing [] in the action name

Commit:
check last guest session, removes local-signup-profile, adds fetch profile with localProfileRepository

Commit:
prevent ensureTeleportAnimation on signup flow

Commit:
remove profile created date in passort

Commit:
formatting

Commit:
updates make watch

Commit:
changes link to discord

Commit:
fixing lint errors

Commit:
Merge remote-tracking branch 'origin/master' into chore/adds_react

Commit:
fixing enable_web3

Commit:
Adapts preview to new singIn

Commit:
minor fixes

Commit:
reverts build-ecs minify format

Commit:
Merge remote-tracking branch 'origin/master' into chore/adds_react

sandrade-dcl committed:
[ProfileHUD] Changes for the new SignIn flow (#1554)

* Move the ManaHUD feature into the ProfileHUD

- Implement ManaCounter into the ProfileHUD.
- Remove all assets related to the old ManaHUD.

* Add "Terms and Conditions" links into the ProfileHUD (#1559)

* [ProfileHUD] Move the Backpack feature into the ProfileHUD (#1564)

* Move the Backpack feature (AvatarEditorHUD) into the ProfileHUD

* Clean unused code and update tests

* Activate LogOut button

* Update text in the Backpack tutorial tooltip

* Fix missing conflicts from the last merge

* [ProfileHUD] Add "SignUp" button into ProfileHUD (#1570)

* Implement SignUp flow from ProfileHUD

* Fix lint errors

* Simplify the redirectToSignUp() function

* Manage visibility of the connected/non-connected wallet sections

* Remove some debug logs

* Update the EmailPrompt delay to 5 minutes

* Visual fix on ProfileHUD

* [ProfileHUD] Edit avatar name (#1578)

* Add button for name edition

* Fix tests

* Implement unverified name edition

* Make the ESC key doesn't save the name edition

* Remove Backpack icon from the taskbar

* [ProfileHUD] Fix LogOut flow (#1588)

* Fix LogOut and SignIn flows (from profileHUD)

* Fix LogOut flow

* Refactorize removeUrlParam() function

* Visual changes on ProfileHUD -> SignIn button

Commit:
updates circleci to deploy on /branch and root for master,staging and release

Commit:
Merge remote-tracking branch 'origin/chore/adds_react' into chore/adds_react

Commit:
Merge remote-tracking branch 'origin/master' into chore/adds_react

BrianAmadori commented on website/src/components/auth/EthLogin.stories.tsx:
```diff
@@ -0,0 +1,40 @@
+};
+
+export const LoginSinging = Template.bind({});
+LoginSinging.args = {
```
```suggestion
export const LoginSigning = Template.bind({});
LoginSigning.args = {
```

BrianAmadori commented on kernel/packages/shared/session/types.ts:
```diff
@@ -1,21 +1,52 @@
+  SING_IN = 'signIn',
+  SING_UP = 'signUp',
+  CONNECT_ADVICE = 'connect_advice',
+  SING_ADVICE = 'sign_advice',
```
```suggestion
  SIGN_IN = 'signIn',
  SIGN_UP = 'signUp',
  CONNECT_ADVICE = 'connect_advice',
  SIGN_ADVICE = 'sign_advice',
```

Commit:
Wording

Commit:
Wording

pravusjif reviewed (approved):
awsome work man!
